c-deps/libroach/engine.cc

```diff
@@ -223,4 +223,62 @@ DBStatus DBImpl::EnvWriteFile(DBSlice path, DBSlice contents) {
   return kSuccess;
 }
 
+// EnvOpenFile opens a new file in the given engine.
+DBStatus DBImpl::EnvOpenFile(DBSlice path, rocksdb::WritableFile** file) {
+  rocksdb::Status s;
+  const rocksdb::EnvOptions soptions;
+  rocksdb::unique_ptr<rocksdb::WritableFile> rocksdb_file;
+
+  // Recursively create the directory of the given file.
+  std::string delimiter = "/";
+  std::string dir = "";
+  std::string rest = ToString(path);
+  int i = 0;
+  int next;
+  while(rest.length() != 0 && (next = rest.find(delimiter)) != -1) {
```

Comment:
make sure to run `make c-deps-fmt`, or `clang-format` directly, or add `clang-format` to your editor save hook for `.cc` and `.h` files.